packages/back-end/src/api/saved-groups/patchSavedGroup.ts

```diff
@@ -0,0 +1,50 @@
+import { PutSavedGroupResponse } from "../../../types/openapi";
+import {
+  getSavedGroupById,
+  toSavedGroupApiInterface,
+  updateSavedGroup,
+} from "../../models/SavedGroupModel";
+import { createApiRequestHandler } from "../../util/handler";
+import { patchSavedGroupValidator } from "../../validators/openapi";
+
+export const patchSavedGroup = createApiRequestHandler(
+  patchSavedGroupValidator
+)(
+  async (req): Promise<PutSavedGroupResponse> => {
+    const { name, values } = req.body;
+    let { owner } = req.body;
+
+    const { id } = req.params;
+
+    const savedGroup = await getSavedGroupById(id, req.organization.id);
+
+    if (!savedGroup) {
+      throw new Error(`Unable to locate the saved-group: ${id}`);
+    }
+
+    if (!values && !name && typeof owner === "undefined") {
+      throw new Error(
+        'You must pass in at least one of the following: "values", "name", "owner".'
+      );
+    }
+
+    if (typeof owner === "string") {
+      owner = owner ? owner : "";
+    } else {
+      owner = savedGroup.owner;
+    }
+
+    const updatedSavedGroup = await updateSavedGroup(id, req.organization.id, {
+      values: values ? values : savedGroup.values,
+      groupName: name ? name : savedGroup.groupName,
+      owner,
+    });
```

Comment:
Is all this necessary? If undefined, wouldn't they not cause updates? How would this behave:

```ts
    const updatedSavedGroup = await updateSavedGroup(id, req.organization.id, req.body);
```

If I'm not mistaken, Zod validation should take care of extraneous properties since we should be using `.strict()` on objects.

Reply:
Currently, this doesn't work for two reasons:

1) Jeremy requested we change the `groupName` field to `name` on this endpoint, but the `updateSavedGroup` method is expecting `groupName`, so doing it this way allows us to rename the property.

2) If we change the body prop back to `groupName` and attempt the snippet above, we get the following error.

<img width="751" alt="Screen Shot 2023-04-18 at 6 53 23 AM" src="https://user-images.githubusercontent.com/75274610/232756610-c169339a-5c1a-4eb2-8f3f-82b9035e78e5.png">